controllers/user.js

```diff
@@ -107,4 +108,36 @@ module.exports = class User {
       res.status(500).json({ error: error });
     }
   }
+  //Followe user
+  static async apiFollowUser(req, res, next) {
+    try {
+      const followerId = req.body.followerId;
+      const followingId = req.body.followingId;
+      if (!followerId || !followingId) {
+        let err = Error("required request info not provided");
+        err.status = 400;
+        throw err;
+      }
+      if (followerId === followingId) {
+        return res.status(400).json({ error: "user should not follow self" });
+      }
+      const follower = await UserService.findOne(followerId);
+      const following = await UserService.findOne(followingId);
+      if (!follower || !following) {
```

Comment:
are you sure the way you use the `!` is correct? all {} is true, and if returning err it is true, error when return it is an object.